migrate.go

```diff
@@ -202,7 +201,7 @@ func EnsureDBVersion(db *sql.DB) (int64, error) {
 		toSkip = append(toSkip, row.VersionId)
 	}
 
-	panic("unreachable")
```

Comment:
I sure agree that migrations may be run in a racy manner, this patch just avoids panic is such a situation.

> I think we should use a distributed lock (think etcd) here to ensure that migrations are run only once.

Note that goose is a general purpose library it should support wide range of databases and dependencies should be kept to minimum, not to mention runtime dependencies.

The proper way to do that is to do is to issue a lock migrations table but first one must ensure such table exists, the way to do that then is to call EnsureDBVersion as it would create the table.